cmd/cosign/cli/sign/sign_blob.go

```diff
@@ -65,7 +67,21 @@ func SignBlobCmd(ro *options.RootOptions, ko options.KeyOpts, regOpts options.Re
 
 	signedPayload := cosign.LocalSignedPayload{}
 
-	if ShouldUploadToTlog(ctx, ko, nil, ko.SkipConfirmation, tlogUpload, "") {
+	if ko.TSAServerURL != "" {
+		clientTSA, err := tsaclient.GetTimestampClient(ko.TSAServerURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create TSA client: %w", err)
+		}
+		b64Sig := []byte(base64.StdEncoding.EncodeToString(sig))
+
+		respBytes, err := tsa.GetTimestampedSignature(b64Sig, clientTSA)
+		if err != nil {
+			return nil, err
+		}
+
+		signedPayload.RFC3161Timestamp = cbundle.TimestampToRFC3161Timestamp(respBytes)
+	}
+	if ShouldUploadToTlog(ctx, ko, nil, ko.SkipConfirmation, tlogUpload, ko.TSAServerURL) {
```

Comment:
I just realized we actually don't need to pass in `ko.SkipConfirmation` or `ko.TSAServerURL` since we pass in `ko` 😓 No need to change, we can do it some other time.

Reply:
yes, it can be simplified. 👍🏻